.github/workflows/tests.yml

```diff
@@ -43,4 +45,4 @@ jobs:
       - name: Run rubocop
         run: bundle exec rake rubocop
       - name: Run tests
-        run: bundle exec rake spec
+        run: bundle exec --gemfile ${{ matrix.gemfile }} rspec
```

Comment:
this makes 100% sure that the gems used are the ones in the specified Gemfile (even though they were the ones installed in a previous step)